### PR TITLE
[#319] Export all packages to OSGi as non-internal

### PR DESCRIPTION
**Pull request by bergmanngabor**
Task-Url: https://github.com/eclipse/eclipse-collections/issues/319
Signed-off-by: bergmann <bergmann@mit.bme.hu>

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
The build timeouts sometimes. I have restarted it. Let's see if it goes through.

File: p2-repository/org.eclipse.collections/osgi.bnd [hunk 1]
```diff
@@ -4,8 +4,6 @@ jcip-annotations-version=${version;===;1.0}
 
 Export-Package: \
  !about.html,!about_files, \
- *.internal*;x-internal:=true;version="${package-version}", \
```
Comment (guw) [inline p2-repository/org.eclipse.collections/osgi.bnd]: Why dropping the `internal` package?

File: p2-repository/org.eclipse.collections/osgi.bnd [hunk 2]
```diff
@@ -4,8 +4,6 @@ jcip-annotations-version=${version;===;1.0}
...
- *.impl*;x-internal:=true;version="${package-version}", \
```
Comment (guw) [inline p2-repository/org.eclipse.collections/osgi.bnd]: @nikhilnanivadekar dropping the `impl` rule is sufficient as the following catch-all rule `*` exports everything using the correct version. However, I would still recommend exporting the `internal` packages with the` x-internal` flag set.

Comment (nikhilnanivadekar) [inline p2-repository/org.eclipse.collections/osgi.bnd]: Makes sense thanks!

**Comment by bergmanngabor** [comment]
@nikhilnanivadekar @guw 

OK, now only *internal packages are marked as internal, and everything is exported.

**Pull request merged**